include/rmm/device_buffer.hpp

```diff
@@ -395,18 +398,25 @@ class device_buffer {
   void set_stream(cuda_stream_view stream) noexcept { _stream = stream; }
 
   /**
-   * @briefreturn{Pointer to the memory resource used to allocate and deallocate}
+   * @briefreturn{The async_resource_ref used to allocate and deallocate}
+   */
+  [[nodiscard]] async_resource_ref memory_resource() const noexcept { return _mr; }
+
+  /**
+   * @brief Enables the `cuda::mr::device_accessible` property
+   *
+   * This property declares that a `device_buffer` provides device accessible memory
    */
-  [[nodiscard]] mr::device_memory_resource* memory_resource() const noexcept { return _mr; }
+  friend void get_property(device_buffer const&, cuda::mr::device_accessible) noexcept {}
```

Comment:
question (non-blocking): We haven't really talked about adding properties to containers yet. Is this something we feel ready to do? I don't think it will be strictly necessary for this PR.

Reply:
@miscco can you comment?  Since this is a pure addition and not a breaking change, the risk is low. However, if we decide to change this in the future perhaps it could be baggage that we with we hadn't added so early?

I would like to understand the motivation for this as well, @miscco .

Reply:
In my mind, giving proper guarantees for the memory allocator is not enough if you drop them on the floor once you allocated. 

So for me it does make sense that the feature that actually uses the memory allocated by the resource ref is also declaring what properties it inherited from the allocator. I am happy to pull it out though

Reply:
I dropped them for now, we can come up with a comprehensive decision later

Reply:
> if you drop them on the floor once you allocated.

Sorry, what do you mean by "drop them on the floor"?  I wasn't saying you should remove this. Just trying to understand how it is used.

Reply:
My point is that currently the properties are solely used on the allocator. Once you have allocated something we do not propagate the properties anymore, but that might actually be the relevant part.

Often someone will pass around a `device_buffer` and it would be awesome to know whether it contains pinned memory without going through the memory resource

Reply:
Yes I agree that would be useful. But this property is hard coded. How would dynamic properties like the example you give work.